patches/minecraft/net/minecraft/block/BlockState.java.patch

```diff
@@ -24,3 +24,12 @@
     public MaterialColor func_185909_g(IBlockReader p_185909_1_, BlockPos p_185909_2_) {
        return this.func_177230_c().func_180659_g(this, p_185909_1_, p_185909_2_);
     }
+@@ -263,7 +267,7 @@
+    }
+ 
+    public List<ItemStack> func_215693_a(LootContext.Builder p_215693_1_) {
+-      return this.func_177230_c().func_220076_a(this, p_215693_1_);
++      return net.minecraftforge.event.ForgeEventFactory.fireBlockDropLootEvent(p_215693_1_, this.func_177230_c().func_220076_a(this, p_215693_1_));
```

Comment:
This patch is wrong, you're hijacking all call to this function as if it's dropping the loot. Where is this function actually used and why are we not patching in it's call site where it actually drops the items? We typically do not patch these bounder methods.

Reply:
I was patching where the drops actually occur before any you suggested I move it somewhere that won't have as many patches. If you have a suggestion as to where this hook should go then let me know, but this is the most direct place to where loot is dropped that I could find without patching every drop method in the block class.

Reply:
As an additional note, this gets called right after the loot is generated. This method in blockstate just calls the method that gets patched in block and then passes data onto some static helpers in the Block class that actually drop the loot into the world. This place makes the most sense to me considering it's where the loot that's going to be dropped actually comes from.